boot/bootutil/src/image_validate.c

```diff
@@ -216,10 +219,19 @@ bootutil_img_validate(struct image_header *hdr, const struct flash_area *fap,
         return -1;
     }
 
+#ifndef APP_mynewt
```

Comment:
Could this be abstracted out into a helper routine which is then called unconditionally? There is starting to be a fair bit of nesting of conditional compilation here, which is getting a bit hard to read and thus audit. (I also wonder if a third RTOS might need its own solution.)

Something like:

```
#if defined(MCUBOOT_SIGN_RSA) || defined(MCUBOOT_SIGN_EC) || defined(MCUBOOT_SIGN_EC256)
#ifndef APP_mynewt
int helper_verify_sig(struct image_header *hdr, uint8_t buf[256], uint8_t hash[32], uint32_t sig_len)
{
     if (hdr->ih_key_id >= bootutil_key_cnt) {
          return -1;
      }
      return bootutil_verify_sig(hash, 32, buf, sig_len, hdr->ih_key_id);
}
#else
int helper_verify_sig(struct image_header *hdr, uint8_t buf[256], uint8_t hash[32], uint32_t sig_len)
{
    int i, rc;
    for (i = 0; i < bootutil_key_cnt; i++) {
        rc = bootutil_verify_sig(hash, sizeof(hash), buf, sig_len, i);
        if (!rc) {
            return rc;
        }
    }
    return 0;
}
#endif
#endif
```

Then call helper_verify_sig() from here.